src/MoBi.Presentation/Presenter/MoBiMainViewPresenter.cs

```diff
@@ -141,5 +147,25 @@ public void Handle(SimulationRunFinishedEvent eventToHandle)
       {
          View.AllowChildActivation = true;
       }
+
+      private void updateWindowTitle()
+      {
+         View.Caption = 
+            string.IsNullOrEmpty(_currentProjectName) 
+               ? _configuration.ProductDisplayName 
+               : $"{_configuration.ProductDisplayName} | {_currentProjectName}";
+      }
+
+      public void Handle(ProjectLoadedEvent eventToHandle)
+      {
+         _currentProjectName = eventToHandle.Project.Name;
+         updateWindowTitle();
+      }
+
+      public void Handle(ProjectClosedEvent eventToHandle)
+      {
+         _currentProjectName = null;
+         updateWindowTitle();
+      }
```

Comment:
Looks great. What about passing the name instead of using a field.

`private void updateWindowTitle(string projectName = null)`

If you agree, then make the change and add Juri.

Reply:
I agree